### PR TITLE
VSCode: Refactor cairols.ts to use `Context` and levelled logging

### PR DESCRIPTION
**Pull request by mkaput**
**Stack**:
- #5015
- #5014
- #5013
- #5012
- #5011
- #4976
- #4975
- #4974
- #4973
- #4959 ⬅
- #4935
- #4934

⚠️ *Part of a stack created by [spr](https://github.com/ejoffe/spr). Do not merge manually using the UI - doing so may have unexpected results.*

- - -
This change is [Reviewable](https://reviewable.io/reviews/starkware-libs/cairo/4959)

### CODE REVIEW

File: vscode-cairo/src/cairols.ts [hunk 1]
```diff
@@ -27,8 +28,8 @@ function findDevLanguageServerAt(
   return findDevLanguageServerAt(path + "/..", depth - 1);
 }
 
-function rootPath(context: vscode.ExtensionContext): string {
-  let rootPath = context.extensionPath;
+function rootPath(ctx: Context): string {
```
Comment (Arcticae) [inline vscode-cairo/src/cairols.ts:31]: Isn't this more like a context function since it uses it?

File: vscode-cairo/src/cairols.ts [hunk 2]
```diff
@@ -143,25 +138,25 @@ async function findScarbExecutablePath(
...
 
-function notifyScarbMissing(outputChannel: vscode.OutputChannel) {
+function notifyScarbMissing(ctx: Context) {
   const errorMessage =
     "This is a Scarb project, but could not find Scarb executable on this machine. " +
     "Please add Scarb to the PATH environmental variable or set the 'cairo1.scarbPath' configuration " +
     "parameter. Otherwise Cairo code analysis will not work.";
-  vscode.window.showWarningMessage(errorMessage);
-  outputChannel.appendLine(errorMessage);
+  void vscode.window.showWarningMessage(errorMessage);
```
Comment (Arcticae) [inline vscode-cairo/src/cairols.ts:146]: Does this show up in 2 different places?

**Review by Arcticae — APPROVED** [review]

**Review by mkaput — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/cairo/4959)* status: 0 of 2 files reviewed, 2 unresolved discussions (waiting on @Arcticae and @orizi)

___
*[`vscode-cairo/src/cairols.ts` line 31 at r1](https://reviewable.io/reviews/starkware-libs/cairo/4959#gh-1481823471:-Nq4DeoSEGW2uTru0COt:b64n9sj) ([raw file](https://github.com/starkware-libs/cairo/blob/0ff66be662a1e7a729718653eec3d11976f1ece1/vscode-cairo/src/cairols.ts#L31)):*
<details><summary><i>Previously, Arcticae (Tomasz Rejowski) wrote…</i></summary><blockquote>

Isn't this more like a context function since it uses it?
</blockquote></details>

It kinda is, but I would rather not touch this code in this PR. In #5013 this function is removed entirely.
___
*[`vscode-cairo/src/cairols.ts` line 146 at r1](https://reviewable.io/reviews/starkware-libs/cairo/4959#gh-1481824945:-Nq4DrhyCMVIkYocivJy:bzhi9it) ([raw file](https://github.com/starkware-libs/cairo/blob/0ff66be662a1e7a729718653eec3d11976f1ece1/vscode-cairo/src/cairols.ts#L146)):*
<details><summary><i>Previously, Arcticae (Tomasz Rejowski) wrote…</i></summary><blockquote>

Does this show up in 2 different places?
</blockquote></details>

Yes, the warning is an alert in UI, while `ctx.log.error` outputs a line in `Output > Cairo Extension` panel.

**Review by Arcticae — COMMENTED** [review]
:lgtm:

> Reviewed 2 of 2 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/cairo/4959#-:-Nq7HIn9DyAbKGqG7Uyy:bnfp4nl)* status: <span>:shipit:</span> complete! all files reviewed, all discussions resolved (waiting on @orizi)

**Comment by mkaput** [comment]
Closing pull request: commit has gone away

**Review by orizi — APPROVED** [review]
:lgtm:

> Reviewed all commit messages.
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/cairo/4959#-:-NqPCUf70UsUxlpxGjjL:bnfp4nl)* status: <span>:shipit:</span> complete! all files reviewed, all discussions resolved (waiting on @mkaput)

**Pull request merged**